File: 240207/바구니 안의 사탕 2/candy-in-the-basket-2.py
n,k=map(int,input().split())
arr=[0]*101

for _ in range(n):
    a,b=map(int,input().split())
    arr[b]=a

ans=0
for c in range(k,100-k):
    cnt=0
    for i in range(c-k,c+k+1):
        cnt+=arr[i]
    ans=max(ans,cnt)

print(ans)

File: 260110/바구니 안의 사탕 2/candy-in-the-basket-2.js
const fs= require("fs");
const input = fs.readFileSync(0).toString().trim().split("\n");
const [n,k] = input[0].split(' ').map(Number);
const candy = input.slice(1).map((c)=>c.split(' ').map(Number));

const arr = Array(101).fill(0);
for(let c of candy){
    const [cnt, pos] = c;
    arr[pos] = cnt;
}

let ans = 0;
for(let i=1; i<100-2*k; i++){
    let cnt = 0;
    for(let j=i; j<=i+2*k; j++){
        cnt+=arr[j];
    }
    ans = Math.max(ans,cnt)
}

console.log(ans)